back-end/processor.py

```diff
@@ -0,0 +1,237 @@
+import sqlite3
```

Comment:
In general, I would recommend breaking up this code into small functions. This would help test-driven development, if you are interested in going down this route. You can use pytest or unittest to run tests that you can write in a directory called tests. This ensures you avoid any bugs and that your data output is what you expect, especially when you want to make changes to your code in the future.

Reply:
I was planning to have smaller functions and wrote a main() function in the first place. But it turned out to be quite linear without much repeatitves lines, and forgot about my plan.

I will mark this down on my to-do list, breaking up the code should help with maintainability and readability.

At the moment, I am just manually testing the code, I will properly include some automated tests in the future.